src/Ormolu/Diff/Text.hs

```diff
@@ -211,8 +209,8 @@ toHunks = go 0 False id id []
       [] ->
         if gotChanges
           then
-            let p = reverse (take margin bothHistory)
-                currentAcc' = addBothAfter p currentAcc
+            let currentAcc' = addBothAfter p currentAcc
+                p = take margin (reverse bothHistory)
```

Comment:
The bug fix in this PR is to change
```haskell
p = reverse (take margin bothHistory)
```
to
```haskell
p = take margin (reverse bothHistory)
```
here. Why is this not necessary below, where we will use the first variant?

Reply:
I have to admit that this `toHunks` function is not the clearest code I have written. In short, the case that I changed is used to pick up lines that are matching in both snippets that being compared when these lines go **after** the differing part. Given that `bothHistory` has the most recent line as its head (so, it is kind of reversed) we should indeed first reverse it back and then take at most `margin` lines:

```
<differing part>
line 1 -----
line 2   |  | take margin (reverse bothHistory)
line 3 --|--
line 4   |
...      | bothHistory, the head is "line N"
line N --
```

the other case is about history lines that precede the differing part:

```
line 1----
line 2    | bothHistory, the head is "line N"
line 3    |
...       |
line N-2 ---
line N-1  | | reverse (take margin bothHistory)
line N------
<differing part>

Reply:
Ah, and we maintain `bothHistory` in reversed form to prevent quadratic time for appending to it. Thanks for the explanation!